config/config.go

```diff
@@ -625,10 +626,23 @@ func (c *RemoteWriteConfig) UnmarshalYAML(unmarshal func(interface{}) error) err
 		}
 	}
 
-	// The UnmarshalYAML method of HTTPClientConfig is not being called because it's not a pointer.
-	// We cannot make it a pointer as the parser panics for inlined pointer structs.
-	// Thus we just do its validation here.
-	return c.HTTPClientConfig.Validate()
+	return c.Validate()
+}
+
+func (c *RemoteWriteConfig) Validate() error {
```

Comment:
Calling this function validate, but leaving much of the validation in `UnmarshalYAML` is not what I expect. I would say either keep it in the unmarshal, or call this function something like `validateClientConfigs`.